integrations/SnapPixel/util.js

```diff
@@ -158,6 +158,41 @@ const ecommEventPayload = (event, message) => {
       };
       break;
     }
+    case "product viewed":
+      payload = {
+        ...payload,
+        item_ids: get(message, "properties.product_id"),
+      };
+      break;
+    case "product list viewed": {
+      let itemIds = [];
+      const products = get(message, "properties.products");
+      if (isDefinedAndNotNull(products)) {
```

Comment:
This should suffice unless that util method is doing something extra.
```suggestion
      if (products) {
```

Reply:
Updated